File: sortcolors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
//  int max =0;
//         for(int i =0 ;i<=nums.size();i++ ){
//             if(nums[i]>nums[i+1]){
//                 max =nums[i];            }
//         }
//         vector<int> x(max,0); 
//             for (int i=0;i<=nums.size();i++){
//        for(int j=0;j<=max;j++){
//            if (nums[i]==j){
//                x[j]=x[j]+1;
               
//            }
//            else{ x[j]=x[j];}
           
           
//        }
//     }  
//         for(int k  : x){
//             cout << x[k];
//         }
         int l=0,mid=0,r=nums.size()-1;
        while(mid<=r)
        {
            if(nums[mid]==1) mid++;
            else if(nums[mid]==0) 
            {
                swap(nums[mid],nums[l]);
                l++;mid++;
            }
            else 
            {
                swap(nums[mid],nums[r]);
                r--;           
            } 
        }

    }
};

File: sort-colors.py
class Solution:
    
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # for i in range(1,len(nums)):
        #     key=nums[i]
        #     j=i-1
        #     while j>=0 and key<nums[j]:
        #         nums[j+1]=nums[j]
        #         j=j-1
        #     nums[j+1]=key
        start =0 
        end=len(nums)-1 
        middle=0
        temp=0
       
        while end>= middle:
            if nums[middle]==0:
               nums[middle],nums[start]=nums[start],nums[middle]
               start=start+1 
               middle=middle +1
            elif nums[middle]==1:
                middle=middle+1
            else:
               nums[middle],nums[end]=nums[end],nums[middle]
               end=end-1    
        return nums